stdlib/mime/src/main/java/org/ballerinalang/mime/util/MultipartDataSource.java

```diff
@@ -86,10 +86,9 @@ public void serialize(OutputStream outputStream) {
      */
     private void serializeBodyPart(OutputStream outputStream, String parentBoundaryString,
                                    ObjectValue parentBodyPart) {
-        final Writer writer = new BufferedWriter(new OutputStreamWriter(outputStream, Charset.defaultCharset()));
         ArrayValue childParts = parentBodyPart.getNativeData(BODY_PARTS) != null ?
                 (ArrayValue) parentBodyPart.getNativeData(BODY_PARTS) : null;
-        try {
+        try (final Writer writer = new BufferedWriter(new OutputStreamWriter(outputStream, Charset.defaultCharset()))){
```

Comment:
You need to instantiate the resources one by one. Refer https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
A good read on the issues with inner resource instantiation https://dzone.com/articles/carefully-specify-multiple-resources-in-single-try

Reply:
Fixed via https://github.com/ballerina-platform/ballerina-lang/pull/40263/commits/01339fd5fc2551466cbb45cb8beb5e2a96f67db0